lib/lotus/entity/dirty_tracking.rb

```diff
@@ -25,46 +25,19 @@ module DirtyTracking
       #   article.changed? # => true
       #
       #   article.changed_attributes # => {:title => "Generation P"}
-      def self.included(base)
-        base.class_eval do
-          extend ClassMethods
-        end
-      end
-
-      module ClassMethods
-        # Override attribute accessors function.
-        # Create setter methods with attribute values checking.
-        # If the new value or a changed, added to @changed_attributes.
-        #
-        # @params attr [Symbol] an attribute name
-        #
-        # @since 0.3.1
-        # @api private
-        #
-        # @see Lotus::Entity::ClassMethods#define_attr_accessor
-        def define_attr_accessor(attr)
-          attr_reader(attr)
 
-          class_eval %{
-            def #{ attr }=(value)
-              _attribute_changed(:#{ attr }, @#{ attr }, value)
-              @#{ attr } = value
-            end
-          }
-        end
-      end
+      attr_reader :init_state
```

Comment:
You're right. But instance variable always public, so it's possible to protect only getters/setters methods.
Maybe you have any suggestions how to store the initial state in immutable way?